packages/xen-api/index.mjs

```diff
@@ -1409,8 +1415,50 @@ export class Xapi extends EventEmitter {
   }
 }
 
+function cachable(fn, getCache) {
+  return async function (...args) {
+    const cache = getCache(args)
+    if (cache === undefined) {
+      return fn.apply(this, args)
+    }
+
+    const key = this.computeCacheKey(...args)
```

Comment:
Yes, this is conform to the specification, these objects are opaque representation of the composite key.

Reply:
But since `cache.has(key)` always seems to return false, it never returns cached values.
I don't understand

Reply:
`args[4]` represents the arguments sent to the XAPI method, e.g., `{ groupName: 'linstor_group/thin_device', hostname: 'xcp-ng-cart13' }`

Since it's an object and `compositeKey` uses it to generate the key, it may be normal that `cache.has(key)` returns `false`. For each request, `args[4]` is a newly created object, even if its properties are the same. So the key will be different each time, right?